### ISSUE TITLE
linter: Categories in config don't have any effect

### ISSUE THREAD

**Issue by Lippiece** [oxc-project/oxc#7721]
### What version of Oxlint are you using?

latest

### What command did you run?

oxlint -c oxlintrc.json --import-plugins

### What does your `.oxlint.json` config file look like?

```json
{
  "categories": {
    "correctness": "error",
    "nursery": "error",
    "pedantic": "error",
    "restriction": "error",
    "style": "error",
    "suspicious": "error"
  }
}
```

### What happened?

My intent is to enable all rules and disable what I don't like.

Running `oxlint -D all --import-plugins` on this code reports 7 errors. Running the command I mentioned reports no warnings or errors.

```ts
import { Hono } from "hono"
import { serveStatic } from "hono/bun"
import { cors } from "hono/cors"
import { etag } from "hono/etag"
import { jwt } from "hono/jwt"
import { logger } from "hono/logger"
import errorHandler from "./functions/errorHandler"
import { FILE_PATHS } from "./functions/generatePathMap"
import api from "./routes/api"

const app           = new Hono()
const jwtWithSecret = jwt({
  secret: process.env.JWT_SECRET!,
})

// Middleware
app
  .use(logger())
  .use(etag())
  .use(
    cors({
      exposeHeaders: ["Access-Control-Allow-Origin"],
      origin       : "*",
    }),
  )

app.use("/api/users", jwtWithSecret)
app.post("/api/(?!(auth)*", jwtWithSecret)
app.put("/api/*", jwtWithSecret)
app.delete("/api/*", jwtWithSecret)

// Routes
app.use("/*", serveStatic({ root: "dist/" }))

app.route("/api", api)

app.get("/sitemap", context => context.json(FILE_PATHS))

// Error
app.onError(errorHandler)

debugger
export type AppType = typeof app
export default app
```
```
.
├── index.ts
└── oxlintrc.json
```

**Comment by Lippiece** [comment]
I'm sorry, it seems Mason neovim registry uses outdated version of the tool. It works in latest.
